internal/configs/testdata/valid-modules/with-tests-backend/test_case_two.tftest.hcl

```diff
@@ -0,0 +1,15 @@
+# This backend is used to set the internal state "foobar-1"
+run "test_1" {
+  state_key = "foobar-1"
+  backend "local" {
+    path = "state/terraform.tfstate"
+  }
+}
+
+# This backend is used to set the internal state "foobar-2"
+run "test_2" {
+  state_key = "foobar-2"
+  backend "local" {
+    path = "state/terraform.tfstate"
+  }
+}
```

Comment:
Should this not be invalid? We have 2 internal states working on the same backend.

Reply:
Indeed, that's now invalid. But the test hasn't failed because the validation is currently [38c284a](https://github.com/hashicorp/terraform/pull/36541/commits/38c284abfdf5e9d2b625ca50425a9672e74bdb78) implemented at a higher level and isn't visible to this test. This is one of my motivations for [seeking feedback here](https://github.com/hashicorp/terraform/pull/36541#discussion_r1975442336), as splitting the validation to different places causes confusion like this.